### ISSUE TITLE
Starcounter service error

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2853]
Such error appears when I start an app in the Playgrounds:

![image](https://cloud.githubusercontent.com/assets/6435556/9463181/d4f61ae6-4acf-11e5-95e4-a8827ad4cb72.png)

In spite of the error everything works fine. The administrator log does not have any entries for this error. Starcounter version: `Polyjuice-DevelopDailyBuilds-2.0.10293`.

@alemoi does this error indicates some invalid xml in the `scnetworkgateway.xml`?

Whatever the reason for the issue is, it should be logged, because this popup blinks and disappears without possibility to investigate the issue and read the full text.

**Comment by alemoi** [comment]
Hard to say, because I am not sure when the gateway configuration is read by the administrator. We should ask @andwah :)

**Comment by andwah** [comment]
Can you check for entries in the Windows Event Viewer?, search for ```Starcounter TrayIcon``` as Source under ```Applications```.

The Starcounter Tray program  tries to get the ```SystemHttpPort```
from the ```Personal.server.config```file. 

As far i know it does not reads from ```scnetworkgateway.xml```

**Comment by miyconst** [comment]
Windows event log says:

```
Starcounter service error
System.Xml.XmlException: Unexpected end of file. Following elements are not closed: acceptingSockets, item, ports, root.
   at System.Xml.XmlExceptionHelper.ThrowXmlException(XmlDictionaryReader reader, String res, String arg1, String arg2, String arg3)
   at System.Xml.XmlBufferReader.GetByteHard()
   at System.Runtime.Serialization.Json.XmlJsonReader.ParseAndSetLocalName()
   at System.Runtime.Serialization.Json.XmlJsonReader.ParseStartElement()
   at System.Runtime.Serialization.Json.XmlJsonReader.Read()
   at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r)
   at System.Xml.Linq.XContainer.ReadContentFrom(XmlReader r, LoadOptions o)
   at System.Xml.Linq.XElement.ReadElementFrom(XmlReader r, LoadOptions o)
   at System.Xml.Linq.XElement.Load(XmlReader reader, LoadOptions options)
   at Codeplex.Data.DynamicJson.Parse(String json, Encoding encoding) in c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.10293.2\Level1\src\Starcounter.XSON\DynamicJson.cs:line 48
   at Starcounter.Tools.Service.Task.NetworkTask.Execute(StarcounterWatcher service, Dictionary`2& appList) in c:\TeamCity\TeamCity8\buildAgent\work\sc-2.0.10293.2\Level1\src\Programs\Starcounter.TrayIcon\Service\NetworkTask.cs:line 79
```

**Comment by miyconst** [comment]
What is the location for this file `Personal.server.config`?

**Comment by warpech** [comment]
It is configured in installer as "Server (databases repository) path". By default it is "C:\Users\<user>\Documents\Starcounter\Personal"

**Comment by miyconst** [comment]
Here is the content of the file, seems to be a valid xml document.

```xml
<?xml version="1.0" encoding="utf-8"?>
<Server xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://schemas.starcounter.com/configuration">
  <DatabaseDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Databases</DatabaseDirectory>
  <EnginesDirectory>C:\Users\Administrator\Documents\Starcounter\Personal</EnginesDirectory>
  <TempDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Temp</TempDirectory>
  <LogDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Logs</LogDirectory>
  <SystemHttpPort>8181</SystemHttpPort>
  <SendUsageAndCrashReports>true</SendUsageAndCrashReports>
  <DefaultDatabaseStorageConfiguration>
    <TransactionLogSize>256</TransactionLogSize>
    <CollationFile>TurboText_en-GB_3.dll</CollationFile>
    <SupportReplication>false</SupportReplication>
  </DefaultDatabaseStorageConfiguration>
  <DefaultDatabaseConfiguration>
    <Runtime>
      <TempDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Temp\[DatabaseName]</TempDirectory>
      <ImageDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Data\[DatabaseName]</ImageDirectory>
      <TransactionLogDirectory>C:\Users\Administrator\Documents\Starcounter\Personal\Data\[DatabaseName]</TransactionLogDirectory>
      <SQLProcessPort>8066</SQLProcessPort>
      <DefaultUserHttpPort>8080</DefaultUserHttpPort>
      <DefaultSessionTimeoutMinutes>20</DefaultSessionTimeoutMinutes>
      <PolyjuiceDatabaseFlag>true</PolyjuiceDatabaseFlag>
      <ChunksNumber>65536</ChunksNumber>
      <SchedulerCount xsi:nil="true" />
    </Runtime>
    <Monitoring>
      <MaxRestartNumber xsi:nil="true" />
      <ResetFailureCountPeriod xsi:nil="true" />
      <GracePeriodAfterConnectionLost xsi:nil="true" />
      <StartupType>Manual</StartupType>
      <MonitoringType>Disabled</MonitoringType>
    </Monitoring>
  </DefaultDatabaseConfiguration>
</Server>
```

**Comment by andwah** [comment]
It looks like the call to ```http://localhost/8181/gw/stats``` do not return the full response body, it seems there is a limit of 64k (or someting close to that) for the response body, 

@alemoi  Can you check if there is a such limit for the response?

**Comment by alemoi** [comment]
Thank you for finding! @andwah yes, current limit is actually 64K. I will try to remove it. Meanwhile you can modify the MAX_STATS_LENGTH in gateway to 256K (1024 * 256), you can push this change.

**Comment by alemoi** [comment]
I rewrote the code so the limit is gone now. The changes are not yet pushed though, but will happen soon.

**Comment by alemoi** [comment]
The complete fix is pushed via b088041e35d69738f4bbc48a72f627bdd262ff81
Now the gateway statistics can be any-size.
Closing the issue.
